File: Day2/onlineStackSpan.py

class StackSpanner:
    def __init__(self):
        self.stack=list()
    def next(self,price):
        ans=1
        while self.stack and self.stack[-1][0]<=price:
            ans+=self.stack.pop()[1]
        self.stack.append([price,ans])
        return ans 
    

File: Day2/onlineStackSpan.cpp
#include<iostream>
#include<bits/stdc++.h>
#include<stack>

using namespace std;
    
int main(){
    stack<pair<int,int>> stack;
    int next(int price) {
        int ans = 1;
        while (!stack.empty() && stack.top().first <= price) {
            ans += stack.top().second;
            stack.pop();
        }
        
        stack.push({price, ans});
        return ans;
    }
    return 0;
}
